src/supertux/game_object.hpp

```diff
@@ -41,6 +42,23 @@ struct GameObjectType
 };
 typedef std::vector<GameObjectType> GameObjectTypes;
 
+/**
+ A helper structure to list all the type_indexes of the classes in the
+ type hierarchy of a given class. This makes it easier to register e.g.
+ a MrIceblock in lists for MrIceBlock, WalkingBadguy, Badguy, Portable,
+ MovingSprite, MovingObject, and GameObject.
+ */
+struct GameObjectClasses
+{
+  std::vector<std::type_index> types;
```

Comment:
The compiler warns that `types` should be initialised.
```
supertux/src/supertux/game_object.hpp: In constructor 'GameObjectClasses::GameObjectClasses()':
supertux/src/supertux/game_object.hpp:51:8: warning: 'GameObjectClasses::types' should be initialized in the member initialization list [-Weffc++]
   51 | struct GameObjectClasses
      |        ^~~~~~~~~~~~~~~~~
supertux/src/supertux/game_object.cpp: In member function 'virtual GameObjectClasses GameObject::get_class_types() const':
supertux/src/supertux/game_object.cpp:111:23: note: synthesized method 'GameObjectClasses::GameObjectClasses()' first required here
  111 |     GameObjectClasses g;
      |                       ^
```

Reply:
This is one of the `-Weffc++` style advice warnings, which recommends initializing _all_ member variables of a struct, since some types do not have default initializers. `std::vector` does. I've avoided the warning by aggregate-initializing `g`.